### PR TITLE
Validate type of experiment ids is an int and type of run ids is uuid

### PR DESCRIPTION
**Pull request by haileyfong**

### CODE REVIEW

File: tests/test_trackingstore.py [hunk 1]
```diff
@@ -1017,3 +1008,41 @@ def test_restore_client_error(mocker):
...
+            "unused-source-type",
+            "unused-source-name",
+            "unused-entry-point-name",
+            RUN_STARTED_AT,
+            "unused-source-version",
+            [MLFLOW_TAG],
+            PARENT_RUN_UUID_HEX_STR,
+        )
+
+    with pytest.raises(ValueError):
```
Comment (acroz) [inline tests/test_trackingstore.py]: We'd usually have separate tests for each of the affected methods, i.e.:

* test_get_experiment_invalid_id
* test_delete_experiment_invalid_id
* test_restore_experiment_invlaid_id
* etc.

File: tests/test_trackingstore.py [hunk 2]
```diff
@@ -1017,3 +1008,41 @@ def test_restore_client_error(mocker):
...
+        store.search_runs(
+            ["invalid-experiment-id", "invalid-experiment-id"], None, None
+        )
+
+
+def test_invalid_run_id(mocker):
```
Comment (acroz) [inline tests/test_trackingstore.py]: Does this test belong in this PR? If so, should probably have tests for all affected methods as above.

File: tests/test_trackingstore.py (hunk 1 above)
Comment (eliasbenussi) [reply tests/test_trackingstore.py]: Yes, good catch, I had not realised how many methods were affected. I have added tests for all cases (I think)

File: mlflow_faculty/trackingstore.py [hunk 3]
```diff
@@ -365,6 +367,11 @@ def search_runs(self, experiment_ids, search_expressions, run_view_type):
 
         try:
             faculty_runs = []
+            experiment_ids = (
+                experiment_ids
```
Comment (acroz) [inline mlflow_faculty/trackingstore.py]:
```suggestion
                None
```

File: tests/test_trackingstore.py [hunk 4]
```diff
@@ -451,28 +469,68 @@ def test_create_run_client_error(mocker):
...
 
+def test_create_run_invalid_parent_run_id(mocker):
+    mock_client = mocker.Mock()
+
+    mocker.patch("faculty.client", return_value=mock_client)
+
+    store = FacultyRestStore(STORE_URI)
+
+    with pytest.raises(ValueError):
+        store.log_batch("invalid-run-id")
```
Comment (acroz) [inline tests/test_trackingstore.py]: ?

**Pull request merged**